From a7552a144f922031a14426b84056fa28c4cc3960 Mon Sep 17 00:00:00 2001
From: Andi Kleen <andi@firstfloor.org>
Date: Sun, 11 Feb 2024 11:31:50 -0800
Subject: [PATCH 1/8] Fix fallback for set_mempolicy_home_node syscall

Correct the syscall number for the fallback.
Add a lot of architectures that support it with the same number to the
ifdef.

Fixes #214
---
 syscall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/syscall.c b/syscall.c
index 63b3e53..a8fe81c 100644
--- a/syscall.c
+++ b/syscall.c
@@ -141,7 +141,7 @@
 
 #if !defined(__NR_set_mempolicy_home_node)
 
-#if defined(__x86_64__) || defined(__aarch64__)
+#if defined(__x86_64__) || defined(__aarch64__) || defined(__i386__) || defined(__powerpc__) || defined(__mips__) || defined(__s390x__)
 #define __NR_set_mempolicy_home_node 450
 #else
 #error "Add syscalls for your architecture or update kernel headers"
-- 
2.41.0

